File: Oauth2Simples/src/test/java/br/com/estudos/oauth2/controller/AuthControllerTest.java
package br.com.estudos.oauth2.controller;

import br.com.estudos.oauth2.dto.login.LoginUserDTO;
import br.com.estudos.oauth2.model.Usuario;
import br.com.estudos.oauth2.service.AuthenticationService;
import br.com.estudos.oauth2.service.StocksDataService;
import br.com.estudos.oauth2.service.StocksService;
import br.com.estudos.oauth2.service.jwt.JwtService;
import lombok.extern.slf4j.Slf4j;
import org.junit.Before;
import org.junit.Test;
import org.junit.jupiter.api.DisplayName;
import org.junit.runner.RunWith;

import org.mockito.InjectMocks;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.mock.mockito.MockBean;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.test.web.servlet.MockMvc;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

@RunWith(SpringJUnit4ClassRunner.class)
//@WebMvcTest(AuthController.class)
@SpringBootTest
public class AuthControllerTest {

    @Autowired
    private MockMvc mockMvc;

    private String jwtToken;

    @Before
    public void setUp(){
        LoginUserDTO dto = new LoginUserDTO();
        dto.setEmail("guih.smi2@gmail.com");
        dto.setPassword("teste");
    }
    @Test
    @DisplayName("Test Spring @Autowired Integration")
    public void testAuthenticateUser() throws Exception {
        mockMvc.perform(post("/auth/login"));
    }

}

File: kotlin/kotlinEstudos/src/test/java/br/com/kotlin/estudos/controller/AuthControllerTest.kt
package br.com.kotlin.estudos.controller

import br.com.kotlin.estudos.services.AuthenticationService
import br.com.kotlin.estudos.services.JwtService
import com.fasterxml.jackson.databind.ObjectMapper
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
import org.springframework.boot.test.mock.mockito.MockBean
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.web.servlet.MockMvc
import br.com.kotlin.estudos.model.dto.LoginUserDTO
import br.com.kotlin.estudos.repository.UsuarioRepository
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.http.MediaType
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.test.web.servlet.post
import org.springframework.web.context.WebApplicationContext
import kotlin.test.assertEquals

@SpringBootTest
@AutoConfigureMockMvc
@ActiveProfiles("test")
class AuthControllerTest {

    @Autowired
    lateinit var context: WebApplicationContext

    @Autowired
    lateinit var objectMapper: ObjectMapper

    @Autowired
    lateinit var authenticationService: AuthenticationService

    @Autowired
    lateinit var jwtService: JwtService

    @Autowired
    private lateinit var mockMvc: MockMvc

    /*@Value()
    private val expiration;*/

    //@Autowired
    //lateinit var jdbcTemplate: JdbcTemplate

    @Test
    fun `should authenticate user and return JWT token`() {
        val loginDto = createLoginDto(email = "guih.smi2@gmail.com", password = "teste")
        val jwtToken = "mocked-jwt-token"
        val expiration = 3600000L // 1 hour
        mockMvc.post("/auth/login") {
            contentType = MediaType.APPLICATION_JSON
            content = toJson(loginDto)
        }.andExpect {
            status { isOk() }
            /*content {
                jsonPath("$.token") { value(jwtToken) }
                jsonPath("$.expiresIn") { value(expiration) }
            }*/
        }
    }

    private fun createLoginDto(email: String, password: String) = LoginUserDTO(
        email = email,
        password = password
    )
    private fun <T> toJson(obj: T): String = objectMapper.writeValueAsString(obj)
}